### PR TITLE
Add notification on /advantage if offers are available

### PR DESCRIPTION
**Pull request by ClementChaumel**
## Done

Added a notification on /advantage to let users know if they have offers available 

## QA

- go to https://ubuntu-com-11159.demos.haus/advantage?test_backend=true
- log in with an account that has offers available
- check that the notification displays correctly
- click the link
- check that it takes you to the offers page

## Screenshots

image

### CODE REVIEW

File: static/js/src/advantage/react/components/Subscriptions/Notifications/Notifications.tsx [hunk 1]
```diff
@@ -28,6 +30,12 @@ const Notifications = () => {
           subscriptions
         </Notification>
       ) : null}
+      {offers?.length > 0 ? (
+        <Notification data-test="offers" inline severity="information">
+          You have one or more Ubuntu Advantage offers to view.{" "}
+          <a href={urls.advantage.offers}>View your offers.</a>
```
Comment (anthonydillon) [inline static/js/src/advantage/react/components/Subscriptions/Notifications/Notifications.tsx:36]: Can you use the actions prop for the notification component? https://canonical-web-and-design.github.io/react-components/?path=/story/notification--actions

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Question in the code

File: static/js/src/advantage/react/components/Subscriptions/Notifications/Notifications.tsx (hunk 1 above)
Comment (ClementChaumel) [reply static/js/src/advantage/react/components/Subscriptions/Notifications/Notifications.tsx]: Done!

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Very small in code comment. Can you add a test for this functionality too?

File: static/js/src/advantage/react/components/Subscriptions/Notifications/Notifications.test.tsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ import {
   userSubscriptionFactory,
   userSubscriptionStatusesFactory,
 } from "advantage/tests/factories/api";
+import { OfferFactory } from "advantage/offers/tests/factories/offers";
 
 describe("Notifications", () => {
```
Comment (anthonydillon) [inline static/js/src/advantage/react/components/Subscriptions/Notifications/Notifications.test.tsx:12]: This is a very generic description of the tests. Could you make it specific to the feature?

File: static/js/src/advantage/react/components/Subscriptions/Notifications/Notifications.test.tsx [hunk 3]
```diff
@@ -46,4 +47,14 @@ describe("Notifications", () => {
     );
     expect(wrapper.find("[data-test='pendingPurchase']").exists()).toBe(false);
   });
+
+  it("displays an offer notification", () => {
```
[the remark this answers is not in the document]
Comment (ClementChaumel) [reply static/js/src/advantage/react/components/Subscriptions/Notifications/Notifications.test.tsx]: Like this?

[the remark this answers is not in the document]
Comment (anthonydillon) [reply static/js/src/advantage/react/components/Subscriptions/Notifications/Notifications.test.tsx]: Yes, makes it easier to see which feature has the error when you end up with lots of tests.

**Pull request merged**